src/Centeva.ObjectStorage/Extensions/ObjectStorageExtensions.cs

```diff
@@ -0,0 +1,38 @@
+﻿namespace Centeva.ObjectStorage.Extensions;
+
+public static class ObjectStorageExtensions
+{
+    public static async Task CopyAsync(this IObjectStorage sourceStorage, StoragePath sourcePath, IObjectStorage targetStorage, StoragePath? targetPath, CancellationToken cancellationToken = default)
+    {
+        targetPath = targetPath?.IsFolder == true
+            ? new StoragePath(StoragePath.Combine(targetPath.Full, sourcePath.Name))
+            : targetPath ?? sourcePath;
+
+        using var sourceStream = await sourceStorage.OpenReadAsync(sourcePath, cancellationToken);
+        await targetStorage.WriteAsync(targetPath, sourceStream!, cancellationToken: cancellationToken);
+    }
+
+    public static async Task CopyAllAsync(this IObjectStorage sourceStorage, StoragePath sourcePath, IObjectStorage targetStorage, StoragePath? targetPath, CancellationToken cancellationToken = default)
```

Comment:
If I make the mentioned change such that we eliminate the concept of "folders" (since they don't apply to blob storage in general), this would then become a "sourcePrefix" and a "targetPrefix", and the copy would replace the source with the target in the "name" of each blob when writing to the new storage provider.  (Or, you could have some kind of mapping Func for more advanced scenarios.)